common.js

```diff
@@ -62,34 +61,63 @@ function languageCheck(language, language_tabs, mutate) {
 
 function getCodeSamples(data) {
     let s = '';
-    let op = data.operation||data.message;
+    const op = data.operation||data.message;
     if (op && op["x-code-samples"]) {
         for (var c in op["x-code-samples"]) {
             var sample = op["x-code-samples"][c];
             var lang = languageCheck(sample.lang, data.header.language_tabs, true);
-            s += '```' + lang + '\n';
-            s += sample.source;
-            s += '\n```\n';
+            s += _generateCodeSnippet(lang, sample.source);
         }
     }
     else {
-        for (let lang of data.header.language_tabs) {
-            let target = lang;
-            if (typeof lang === 'object') target = Object.keys(target)[0];
-            let lcLang = languageCheck(target,data.header.language_tabs,false);
-            var templateName = 'code_' + lcLang.substring(lcLang.lastIndexOf('-') +
-1);
-            var templateFunc = data.templates[templateName];
-            if (templateFunc) {
-                s += '```'+lcLang+'\n';
-                s += templateFunc(data)+'\n';
-                s += '```\n\n';
-            }
-        }
+        const samplesGenerator = data.options.httpsnippet
+            ? httpsnippetGenerator.generate
+            : _fileTemplateGenerator;
+
+        const codeSamples = data.header.language_tabs
+            .map(tab => {
+                const lang = typeof tab === 'object'
+                    ? Object.keys(tab)[0]
+                    : tab;
+
+                const lowerCaseLanguage = languageCheck(lang, data.header.language_tabs, false);
+                const target = _getLanguageTarget(lowerCaseLanguage);
+                const client = _getLanguageClient(lang, data.options.language_clients);
+
+                const sample = (target && samplesGenerator(target, client, data)) || '';
+                return (sample && _generateCodeSnippet(lowerCaseLanguage, sample)) || '';
+            });
+
+        s += codeSamples.join('');
     }
     return s;
 }
 
+function _getLanguageTarget(lang) {
```

Comment:
Style point: is there an advantage to the leading underscore. 'Private' functions are already indicated by not being included in the `module.exports`.

Reply:
Just immediate feedback without having to check what is actually exported.
Changed to remove them for keeping consistency with rest of code.